posts/2024-xx-xx-spring-boot-3.adoc

```diff
@@ -0,0 +1,223 @@
++
+5. Stop the server (with  `<Ctrl>+C` or `q,<Enter>` to quit dev mode).
++
+6. Recap
++
+The app is bootstrapped by the web container via the ServletInitializer, extending `SpringBootServletInitializer`.  We go out of our way to run this example with only default Liberty and Maven configuration even though it'd be more typical to maintain a server.xml configuration for your app.
+
+== Use Liberty Spring Boot feature deployment
+
+We're going to reuse the app from the previous example.
+
+1. Add liberty-maven-plugin with Spring Boot deployment config to pom.xml:
++
+[source,xml]
+----
+    <plugin>
+        <groupId>io.openliberty.tools</groupId>
+        <artifactId>liberty-maven-plugin</artifactId>
+        <version>3.10</version>
+        <configuration>
+            <appsDirectory>apps</appsDirectory>
+            <deployPackages>spring-boot-project</deployPackages>
+        </configuration>
+    </plugin>
+----
++
+2. Create your Liberty server.xml by copying from the Spring Boot template
++
+[source,sh]
+----
+./mvnw clean liberty:create -Dtemplate=springBoot3
+mkdir -p src/main/liberty/config/
+cp target/liberty/wlp/usr/servers/defaultServer/server.xml src/main/liberty/config/server.xml
+----
++
+3. Add application arguments to your Spring Boot application configuration in Liberty server.xml
++
+[source,xml]
+----
+ <springBootApplication id="spring-liberty" location="thin-spring-liberty-0.0.1-SNAPSHOT.war" name="spring-liberty">
+    <applicationArgument>--server.servlet.context-path=/testpath</applicationArgument>
+    <applicationArgument>--myarg=myval</applicationArgument>
+ </springBootApplication>
+----
++
+4. Add println to echo args in output
++
+[source,java]
+----
+	public static void main(String[] args) {
+        System.out.println("Arguments array = " + java.util.Arrays.toString(args));
+		SpringApplication.run(SpringLibertyApplication.class, args);
+	}
+----
++
+5. Run
++
+Though we won't benefit from using dev mode, we can still use the "lifecycle" built into the 'run' goal.  We do need to do a 'package' first to get the Spring Boot "repackage" packaging.
++
+[source,sh]
+----
+./mvnw package liberty:run
+----
++
+6.  Observe the output:
++
+[source,sh]
+----
+[INFO] Arguments array = [--server.servlet.context-path=/testpath, --myarg=myval]
+[INFO]   .   ____          _            __ _ _
+[INFO]  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
+[INFO] ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
+[INFO]  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
+[INFO]   '  |____| .__|_| |_|_| |_\__, | / / / /
+[INFO]  =========|_|==============|___/=/_/_/_/
+[INFO]  :: Spring Boot ::                (v3.2.4)
+----
++
+and also
++
+[source,sh]
+----
+[INFO] [AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/testpath/
+----
++
+7. Test the application by visiting the following URL in a browser: http://localhost:9080/testpath/
```

Comment:
What should you see at that URL?

Reply:
TODO - I owe you an answer.  Will set up and run

Reply:
DONE